Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,922 +22,340 @@
+					const carve::math::Matrix& pos,
+					EXPRESSReference<carve::input::PolylineSetData>& polylineData) {
 
-							IfcBoundedSurface						http://www.buildingsmart-tech.org/ifc/IFC4x1/final/html/schema/ifcgeometryresource/lexical/ifcboundedsurface.htm
-								IfcBSplineSurface*
-								IfcCurveBoundedPlane
-								IfcCurveBoundedSurface*
-								IfcRectangularTrimmedSurface*
+					// (1/3) IfcBoundedSurface SUBTYPE of IfcSurface
+					if (surface.isOfType<typename IfcEntityTypesT::IfcBoundedSurface>()) {
+						return convertIfcBoundedSurface(surface.as<typename IfcEntityTypesT::IfcBoundedSurface(), pos, polylineData);
+					}
 
-							IfcElementarySurface					http://www.buildingsmart-tech.org/ifc/IFC4x1/final/html/schema/ifcgeometryresource/lexical/ifcelementarysurface.htm
-								IfcCylindricalSurface*
-								IfcPlane
-								IfcSphericalSurface*
-								IfcToroidalSurface*
+					// (2/3) IfcElementarySurface SUBTYPE of IfcSurface
+					if (surface.isOfType<typename IfcEntityTypesT::IfcElementarySurface>()) {
+						return convertIfcElementarySurface(surface.as<typename IfcEntityTypesT::IfcElementarySurface(), pos, polylineData);
+					}
 
-							IfcSweptSurface							http://www.buildingsmart-tech.org/ifc/IFC4x1/final/html/schema/ifcgeometryresource/lexical/ifcsweptsurface.htm
-								IfcSurfaceOfLinearExtrusion*
-								IfcSurfaceOfRevolution*
+					// (3/3) IfcSweptSurface SUBTYPE of IfcSurface
+					if (surface.isOfType<typename IfcEntityTypesT::IfcSweptSurface>()) {
+						return convertIfcSweptSurface(surface.as<typename IfcEntityTypesT::IfcSweptSurface(), pos, polylineData);
+					}
+				}
 
-							*: not implemented in FaceConverter.h
+				//--------------------------------------------------------------------------------------------
+				// IfcBoundedSurface
+				//--------------------------------------------------------------------------------------------
 
+					/*! \brief  Converts \c IfcBoundedSurface by calling the conversion function according to subtype.
+					\param		surface	\c IfcBoundedSurface entity to be interpreted.
+					\param		pos
+					\param		polylineData
+					\return
+					\note		The \c IfcBoundedSurface subtypes are: \c IfcBSplineSurface, \c IfcCurveBoundedPlane, \c IfcCurveBoundedSurface and \c IfcRectangularTrimmedSurface.
 					*/
 
-					// ************************************************************************************************************************	//
-					//	IfcBoundedSurface SUPTYPE of IfcSurface																					//
-					//	ABSTRACT SUPERTYPE of IfcBSplineSurface, IfcCurveBoundedPlane, IfcCurveBoundedSurface, IfcRectangularTrimmedSurface		//
-					// ************************************************************************************************************************	//
-
-					std::shared_ptr<typename IfcEntityTypesT::IfcBoundedSurface> bounded_surface = std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcBoundedSurface>(surface);
-
-					if(bounded_surface) {

```

Comment:
None of the functionalities can be lost. Pack it in another function, yes, but do not delete code that is there.